collect/collect.go

```diff
@@ -515,7 +515,7 @@ func (i *InMemCollector) processSpan(sp *types.Span) {
 	trace.AddSpan(sp)
 
 	// if this is a root span, send the trace
-	if isRootSpan(sp) {
+	if isRootSpan(sp, i.Config) {
```

Comment:
instead of passing config in as a parameter, which feels inappropriate here, please make `isRootSpan` into a member function of InMemCollector and simply reference i.Config inside the function.